### PR TITLE
feat: factorize pagination

### PR DESCRIPTION
**Pull request by gfyrag**

### CODE REVIEW

File: components/ledger/pkg/storage/accounts.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
+package storage
+
+import (
+	"github.com/formancehq/stack/libs/go-libs/metadata"
+)
+
+type AccountsQuery OffsetPaginatedQuery[AccountsQueryPayload]
+
+type AccountsQueryPayload struct {
+	AfterAddress    string
```
Comment (paul-nicolas) [inline components/ledger/pkg/storage/accounts.go:10]: Should we add json tags on this struct since it will be the one unmarshaled from the url ?

Comment (paul-nicolas) [reply components/ledger/pkg/storage/accounts.go:10]: Some for other structs

File: components/ledger/pkg/storage/accounts.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+		Order:    OrderAsc,
+		Payload: AccountsQueryPayload{
+			Metadata: metadata.Metadata{},
+		},
+	}
+}
+
+func (a AccountsQuery) WithPageSize(pageSize uint64) AccountsQuery {
+	if pageSize != 0 {
+		a.PageSize = pageSize
```
Comment (paul-nicolas) [inline components/ledger/pkg/storage/accounts.go:29]: Should we return an error when pageSize == 0 ? Maybe not here but in the caller ?

File: components/ledger/pkg/storage/accounts.go (hunk 1 above)
Comment (gfyrag) [reply components/ledger/pkg/storage/accounts.go:10]: Good point. I will add them.

File: components/ledger/pkg/storage/sqlstorage/ledger/accounts.go [hunk 3]
```diff
@@ -30,58 +29,34 @@ type Accounts struct {
...
-		metadata        = p.Filters.Metadata
-		balance         = p.Filters.Balance
-		balanceOperator = p.Filters.BalanceOperator
-	)
-
-	if address != "" {
-		sb.Where("address ~* ?", "^"+address+"$")
-		t.AddressRegexpFilter = address
+
+	if p.Payload.Address != "" {
```
Comment (paul-nicolas) [inline components/ledger/pkg/storage/sqlstorage/ledger/accounts.go:35]: Payload is way less understandable than Filters, would it be possible to change it easily ?

**Review by paul-nicolas — APPROVED** [review]

**Pull request merged**